### PR TITLE
New AWS module for ec2 internet gateway facts

### PR DESCRIPTION
**Pull request by naslanidis**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

ec2_vpc_igw_facts
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel f39ad1f13a) last updated 2016/05/01 14:01:36 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible']

```
##### SUMMARY

This module returns internet gateway facts for an AWS account.

```
TASK [print output] ************************************************************
ok: [localhost] => {
    "msg": {
        "changed": false, 
        "internet_gateways": [
            {
                "attachments": [
                    {
                        "state": "available", 
                        "vpc_id": "vpc-02642b23"
                    }
                ], 
                "internet_gateway_id": "igw-12e3645d", 
                "tags": [
                    {
                        "key": "Name", 
                        "value": "igw-test"
                    }
                ]
            }
        ]
    }
}
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @naslanidis for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/ec2_vpc_igw_facts.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    type: list
+
+changed:
+    description: True if listing the internet gateways succeeds
+    type: bool
+    returned: always
+'''
+
+try:
+    import json
```
Comment (wimnat) [inline cloud/amazon/ec2_vpc_igw_facts.py]: json shouldn't be in this try

File: cloud/amazon/ec2_vpc_igw_facts.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+    try:
+        all_internet_gateways = client.describe_internet_gateways(**params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg=str(e))
+
+    for internet_gateway in all_internet_gateways['InternetGateways']:
+        all_internet_gateways_array.append(get_internet_gateway_info(internet_gateway))
+
+    module.exit_json(internet_gateways=all_internet_gateways_array)
```
Comment (wimnat) [inline cloud/amazon/ec2_vpc_igw_facts.py]: You need to change the boto result to be ansible friendly by changing from CamelCase to snake_case.

Use the helper function 'camel_dict_to_snake_dict'.  See GUIDELINES and ec2_snapshot_facts for example

**Comment by wimnat** [comment]
needs_revision

**Comment by naslanidis** [comment]
Hi @wimnat 

I'm not sure what CamelCase you're referring to. 

The InternetGateways is the boto3 API return syntax:

```
Response Syntax

{
'InternetGateways': [
    {
        'InternetGatewayId': 'string',
        'Attachments': [
            {
                'VpcId': 'string',
                'State': 'attaching'|'attached'|'detaching'|'detached'
            },
        ],
        'Tags': [
            {
                'Key': 'string',
                'Value': 'string'
            },
        ]
    },
]
}
```

http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.Client.describe_internet_gateways

Or are you referring to something else?

**Comment by wimnat** [comment]
Yep so boto favours CamelCase but ansible standard is snake_case so that
helper function helps to convert the boto result

**Comment by gregdek** [comment]
Thanks @naslanidis for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by naslanidis** [comment]
ready_for_review

**Comment by naslanidis** [comment]
@wimnat 

I was using 2.1 previously which didn't have the conversion function available. 

I've changed to 2.2 and have included the snake_case conversion.

**Comment by gregdek** [comment]
Thanks @naslanidis for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by kalyat** [comment]
We have been using it for months now. shipit

**Comment by ljudzia** [comment]
Been very useful. shipit

**Comment by naslanidis** [comment]
has been migrated to ansible repo

**Comment by willthames** [comment]
Needs fixing for python 3 so that it passes all tests, but will be pushing for this once tests pass

**Comment by naslanidis** [comment]
@willthames 

I have a bunch I need to update for Python 3. I will do this during the coming weekend hopefully.

**Pull request closed without merging**